data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,258 @@
+  }
+  def druidAlgorithm(reportConfig: ReportConfig)(implicit spark: SparkSession, sqlContext: SQLContext, fc: FrameworkContext, sc:SparkContext, config: JobConfig): RDD[DruidOutput]  ={
+    val queryDims = reportConfig.metrics.map { f =>
+      f.druidQuery.dimensions.getOrElse(List()).map(f => f.aliasName.getOrElse(f.fieldName))
+    }.distinct
+    if (queryDims.length > 1) throw new DruidConfigException("Query dimensions are not matching")
+
+    val interval = reportConfig.dateRange
+    val granularity = interval.granularity
+    val reportInterval = if (interval.staticInterval.nonEmpty) {
+      interval.staticInterval.get
+    } else if (interval.interval.nonEmpty) {
+      interval.interval.get
+    } else {
+      throw new DruidConfigException("Both staticInterval and interval cannot be missing. Either of them should be specified")
+    }
+    val metrics = reportConfig.metrics.map { f =>
+
+      val queryInterval = if (interval.staticInterval.isEmpty && interval.interval.nonEmpty) {
+        val dateRange = interval.interval.get
+        getDateRange(dateRange, interval.intervalSlider,f.druidQuery.dataSource)
+      } else
+        reportInterval
+
+      val queryConfig = if (granularity.nonEmpty)
+        JSONUtils.deserialize[Map[String, AnyRef]](JSONUtils.serialize(f.druidQuery)) ++ Map("intervalSlider" -> interval.intervalSlider,
+          "intervals" -> queryInterval, "granularity" -> granularity.get)
+      else
+        JSONUtils.deserialize[Map[String, AnyRef]](JSONUtils.serialize(f.druidQuery)) ++ Map("intervalSlider" -> interval.intervalSlider,
+          "intervals" -> queryInterval)
+      val data = DruidDataFetcher.getDruidData(JSONUtils.deserialize[DruidQueryModel](JSONUtils.serialize(queryConfig)))
+      data.map { x =>
+        val dataMap = JSONUtils.deserialize[Map[String, AnyRef]](x)
+        val key = dataMap.filter(m => (queryDims.flatten).contains(m._1)).values.map(f => f.toString).toList.sorted(Ordering.String.reverse).mkString(",")
+        (key, dataMap)
+
+      }
+    }
+    val finalResult = metrics.fold(sc.emptyRDD)(_ union _)
+    finalResult.map { f =>
+      DruidOutput(f._2)
+    }
+  }
+
+  def druidPostProcess(data: RDD[DruidOutput],request_id: String,reportConfig: ReportConfig, storageConfig:StorageConfig)(implicit spark: SparkSession, sqlContext: SQLContext, fc: FrameworkContext, sc:SparkContext, config: JobConfig): OnDemandDruidResponse = {
+    val labelsLookup = reportConfig.labels
+    implicit val sqlContext = new SQLContext(sc)
+    val dimFields = reportConfig.metrics.flatMap { m =>
+      if (m.druidQuery.dimensions.nonEmpty) m.druidQuery.dimensions.get.map(f => f.aliasName.getOrElse(f.fieldName))
+      else if(m.druidQuery.sqlDimensions.nonEmpty) m.druidQuery.sqlDimensions.get.map(f => f.fieldName)
+      else List()
+    }
+    val dataCount = sc.longAccumulator("DruidReportCount")
+    val filesMutable = scala.collection.mutable.MutableList[String]()
```

Comment:
As updated earlier, this filesMutable variable can be avoided.

Reply:
Fixed this logic